### PR TITLE
Warehouse FAQ updates

### PR DESCRIPTION
**Pull request by ameypv-rudder**
## Description of the change

> Added new warehouse FAQ.

## Type of documentation

- [ ] New documentation
- [x] Documentation update
- [ ] Hotfix

## Notion ticket link

> [Warehouse FAQ](https://www.notion.so/rudderstacks/Warehouse-FAQ-updates-6964f7839d9a4db8bdf10860cd64f9ad)

### CODE REVIEW

File: docs/data-warehouse-integrations/gcs-datalake.mdx [hunk 1]
```diff
@@ -153,6 +153,16 @@ If you have your deployment in the EU region, you can allowlist only the followi
...
 
+## FAQ
+
+### Why am I observing files written in the location `<namespace>/rudder-warehouse-staging-logs/`?
+
+RudderStack collects all the unprocessed data flowing to the warehouse destinations as staging files. It stores these files in the object storage at the location `https://storage.googleapis.com/rudder-warehouse-staging-logs/`. 
+
+Once the staging files are processed, RudderStack separates them by the event name and sends them to the specified destination in the following format:
+
+`https://storage.googleapis.com/<bucketName>/<prefix>/rudder-datalake/<namespace>/<tableName>/<table_suffix>/YYYY/MM/DD/HH`.
```
Comment (Jayachand) [inline docs/data-warehouse-integrations/gcs-datalake.mdx:164]: lets remove `<table_suffix>/YYYY/MM/DD/HH`. Haven't released yet

File: docs/data-warehouse-integrations/warehouse-faqs.mdx [hunk 2]
```diff
@@ -24,54 +52,21 @@ uploadFreqInS=1800 #This field lets you control syncPeriod if above field set to
...
+## Staging
 
-<div class="infoBlock">
+### Why am I observing files written in the location `<namespace>/rudder-warehouse-staging-logs/`?
 
-Refer to the warehouse-specific destination settings to configure a namespace in the RudderStack dashboard.
-</div>
-
-### How can I achieve identity resolution with multiple sources connected to one warehouse destination?
+RudderStack collects all the unprocessed data flowing to the warehouse destinations as staging files. It stores these files in the object storage at the location `/rudder-warehouse-staging-logs/`.
```
Comment (Jayachand) [inline docs/data-warehouse-integrations/warehouse-faqs.mdx:59]: Do you think we should add location as `<namespace>/rudder-warehouse-staging-logs/`

File: docs/data-warehouse-integrations/gcs-datalake.mdx (hunk 1 above)
Comment (ameypv-rudder) [reply docs/data-warehouse-integrations/gcs-datalake.mdx:164]: oh okay - i saw [this comment](https://rudderlabs.slack.com/archives/C02MQ9640H1/p1653653697352479?thread_ts=1653571353.574159&cid=C02MQ9640H1) and thought of adding the info for clarity. I'll remove this till we push the feature.

Comment (ameypv-rudder) [inline docs/data-warehouse-integrations/gcs-datalake.mdx:164]:
```suggestion
`https://storage.googleapis.com/<bucketName>/<prefix>/rudder-datalake/<namespace>/<tableName>`.
```

File: docs/data-warehouse-integrations/warehouse-faqs.mdx (hunk 2 above)
Comment (ameypv-rudder) [inline docs/data-warehouse-integrations/warehouse-faqs.mdx:59]:
```suggestion
RudderStack collects all the unprocessed data flowing to the warehouse destinations as staging files. It stores these files in the object storage at the location `<namespace>/rudder-warehouse-staging-logs/`. 
```

File: docs/data-warehouse-integrations/warehouse-faqs.mdx [hunk 3]
```diff
@@ -294,6 +327,29 @@ products_i_phone_version : 11
...
+
+To set your preferred data type for a particular column, it is highly recommended to follow these steps:
+
+1. Create a column in the warehouse with a dummy name and the required data type.
+2. Cast the data from the original column and load it into the dummy column.
+3. Drop the original column.
+4. Rename the dummy column to the original column name.
+
+<div class="dangerBlock">
+During steps 3 and 4, the tables will be in a locked state. This might impact real-time data uploads/syncs.
```
Comment (Jayachand) [inline docs/data-warehouse-integrations/warehouse-faqs.mdx:348]: Please mention this option: 
`If above operations are taking longer time, you can halt warehouse operations for time being. More at: https://www.rudderstack.com/docs/data-warehouse-integrations/warehouse-faqs/#how-do-i-avoid-lags-in-data-syncs-while-heavy-loads-are-running-on-my-datawarehouse`

File: docs/data-warehouse-integrations/warehouse-faqs.mdx [hunk 4]
```diff
@@ -22,54 +49,21 @@ uploadFreqInS=1800 #This field lets you control syncPeriod if above field set to
 
 You can view the warehouse upload status in the **Live Events** section of the destination in the RudderStack dashboard.
 
-### Can I change the namespace (schema name) of my data warehouse in RudderStack?
+## Staging
 
-Yes, you can. Although the default namespace will be the source name with some modifications to it, RudderStack also provides an option in the dashboard to explicitly set the namespace of the dataset in the warehouse.
+### Why am I observing files written in the location `<namespace>/rudder-warehouse-staging-logs/`?
```
Comment (ishachopra0401) [inline docs/data-warehouse-integrations/warehouse-faqs.mdx:54]:
```suggestion
### What are these files being written in the location `<namespace>/rudder-warehouse-staging-logs/`?
```

File: docs/data-warehouse-integrations/warehouse-faqs.mdx [hunk 5]
```diff
@@ -225,6 +226,31 @@ The table name truncation logic is as follows:
 - For **Data Lake** destinations (**S3 Data Lake, GCS Data Lake, Azure Data Lake**), there is no set limit.
 - For **other** destinations, the table name is truncated after 127 characters.
 
+### Can I change the name of the table in which my events are synced?
+
+Yes, you can add a <Link to="">transformation</Link> to change the event name and, consequently, the name of the table in which the events are synced.
```
Comment (ishachopra0401) [inline docs/data-warehouse-integrations/warehouse-faqs.mdx:231]: hyperlink to be added

File: docs/data-warehouse-integrations/warehouse-faqs.mdx [hunk 6]
```diff
@@ -292,6 +318,30 @@ products_i_phone_version : 11
 products_android_version : 16
 ```
 
+### How does RudderStack determine the column data type? Can I change an existing data type for a column?
+
+RudderStack determines the columns' data type based on the values contained in the first event when it is uploaded (during the first upload sync).
```
Comment (ishachopra0401) [inline docs/data-warehouse-integrations/warehouse-faqs.mdx:323]:
```suggestion
RudderStack determines the data type of a column based on its value in the first event (during the first upload sync).
```

File: docs/data-warehouse-integrations/gcs-datalake.mdx [hunk 7]
```diff
@@ -151,6 +151,16 @@ If you have your deployment in the EU region, you can allowlist only the followi
 All the outbound traffic is routed through these RudderStack IPs.
 </div>
 
+## FAQ
+
+### Why am I observing files written in the location `<namespace>/rudder-warehouse-staging-logs/`?
```
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply docs/data-warehouse-integrations/gcs-datalake.mdx:156]: @ishachopra0401 slightly tweaked your suggestion. LMK if it makes sense.

File: docs/data-warehouse-integrations/warehouse-faqs.mdx (hunk 5 above)
Comment (ameypv-rudder) [inline docs/data-warehouse-integrations/warehouse-faqs.mdx:231]:
```suggestion
Yes, you can add a <Link to="/transformations/">transformation</Link> to change the event name and, consequently, the name of the table in which the events are synced. 
```

**Review by ishachopra0401 — APPROVED** [review]

**Pull request merged**